tests/program/linspace_tests.py

```diff
@@ -63,6 +63,81 @@ def test_output(self):
         assert_vm_output_almost_equal(self, self.output, vm.history)
 
 
+class PrePostDepTest(TestCase):
+    def setUp(self):
+        hold = ConstantPT(10 ** 6, {'a': '-1. + idx * 0.01'})
+        hold_random = ConstantPT(10 ** 5, {'a': -.4})
+        # self.pulse_template = (hold_random@(hold_random@hold).with_repetition(10)@hold_random@hold)\
+        self.pulse_template = (hold_random @ hold.with_repetition(10)).with_iteration('idx', 200)
+
+        self.program = LinSpaceIter(
+            length=200,
+            body=(
+                LinSpaceHold(bases=(-.4,), factors=(None,), duration_base=TimeType(10**5), duration_factors=None),
+                LinSpaceRepeat(body=(
+                LinSpaceHold(bases=(-1.,),factors=((0.01,),),duration_base=TimeType(10**6),duration_factors=None),
+                ), count=10),
+                # LinSpaceHold(bases=(-.4),factors=None,duration_base=TimeType(10**6),duration_factors=None),
+                # LinSpaceHold(bases=(-1.,),factors=((0.01,),),duration_base=TimeType(10**6),duration_factors=None)
+            ),)
+        
+
+        key = DepKey.from_voltages((0.01,), DEFAULT_INCREMENT_RESOLUTION)
+
+        self.commands = [
+            Set(channel=0, value=-0.4, key=DepKey(factors=())),
+             Wait(duration=TimeType(100000, 1)),
+             #here is what currently happens:
+             Set(channel=0, value=-1.0, key=DepKey(factors=(10000000,))),
+             Wait(duration=TimeType(1000000, 1)),
+             LoopLabel(idx=0, count=9),
+             Wait(duration=TimeType(1000000, 1)),
+             LoopJmp(idx=0),
+             #however, i think this is what should happen (maybe also with an additional "Set" before,
+             #which might cause complications if omitted in other contexts like AWG amplitude:
+             #LoopLabel(idx=0, count=10),
+             #Set(channel=0, value=-1.0, key=DepKey(factors=(10000000,))),
+             #Wait(duration=TimeType(1000000, 1)),
+             #LoopJmp(idx=0),
+             
+             LoopLabel(idx=1, count=199),
+             Set(channel=0, value=-0.4, key=DepKey(factors=())),
+             Wait(duration=TimeType(100000, 1)),
+             Increment(channel=0, value=0.01, dependency_key=DepKey(factors=(10000000,))),
+             Wait(duration=TimeType(1000000, 1)),
+             LoopLabel(idx=2, count=9),
+             #also here, an increment 0 may be helpful (at least to be able to force).
```

Comment:
The increment is not emitted due to an optimization. Why would it be helpful?

Reply:
I would go further and replace loops that are constant by waits

Reply:
for wait it's not helpful. Just for other things, like WF scale, the program would need to know about the register it has to choose, which may work smoother if the corresponding Set command with the relevant DepKey is directly in front of it. Or maybe not, I dont' know.

Reply:
I think a scaled play should have an dependency key.